analytic_engine/src/sst/parquet/async_reader.rs

```diff
@@ -51,12 +56,67 @@ use crate::sst::{
 
 type SendableRecordBatchStream = Pin<Box<dyn Stream<Item = Result<ArrowRecordBatch>> + Send>>;
 
+pub type AsyncFileReaderRef = Arc<dyn AsyncFileReader>;
+
+#[async_trait]
+pub trait AsyncFileReader: Send + Sync {
+    async fn file_size(&self) -> GenericResult<usize>;
+
+    async fn get_byte_range(&self, range: Range<usize>) -> GenericResult<Bytes>;
+
+    async fn get_byte_ranges(&self, ranges: &[Range<usize>]) -> GenericResult<Vec<Bytes>>;
+}
+
+/// Fetch and parse [`ParquetMetadata`] from the file reader.
+///
+/// Referring to: https://github.com/apache/arrow-datafusion/blob/ac2e5d15e5452e83c835d793a95335e87bf35569/datafusion/core/src/datasource/file_format/parquet.rs#L390-L449
+async fn fetch_parquet_metadata_from_file_reader(
```

Comment:
How about move this function to parquet_ext component?

Reply:
It sounds great.